pages/docs/reference/scope-functions.md

```diff
@@ -0,0 +1,530 @@
+```kotlin
+fun main() {
+    val str = "Hello"
+    // this
+    str.run {
+        println("The receiver string's length is $length")
+        //println("The receiver string's length is ${this.length}") // does the same
+    }
+
+    // it
+    str.let {
+        println("The receiver string's length is ${it.length}")
+    }
+}
+```
+</div>
+
+#### this
+
+`run`, `with`, and `apply` refer to the context object as a lambda receiver - by keyword `this`. Hence, in their lambdas, the object is available as it would be in ordinary class functions. In most cases, you can omit `this` when accessing the members of the receiver object, making the code shorter. On the other hand, if `this` is omitted, it can be hard to distinguish between the receiver members and external objects or functions. So, having the context object as a receiver (`this`) is recommended for lambdas that mainly operate the object members: call its functions or assign properties.
+
+<div class="sample" markdown="1" theme="idea">
+```kotlin
+data class Person(var name: String, var age: Int = 0, var city: String = "")
+
+fun main() {
+//sampleStart
+    val adam = Person("Adam").apply { 
+        age = 20                       // same as this.age = 20 or adam.age = 20
+        city = "London"
+    }
+//sampleEnd
+}
+```
+</div>
+
+#### it
+
+In turn, `let` and `also` have the context object as a lambda argument. If the argument name is not specified, the object is accessed by the implicit default name `it`. `it` is shorter than `this` and expressions with `it` are usually easier for reading. However, you can't omit the argument name, so there is no way to shorten calls to the object functions or properties. Hence, having the context object as `it` is better when the object is mostly used as an argument in function calls. `it` is also better if you operate multiple variables in the code block.
+
+<div class="sample" markdown="1" theme="idea">
+```kotlin
+fun writeToLog(str: String) {}
+fun writeToFile(str: String) {}
+
+fun main() {
+//sampleStart
+    "Hello".also {
+        println("Also called on the string $it")
+        writeToLog(it)
+        writeToFile(it)
+    }
+//sampleEnd
+}
+```
+</div>
+
+Additionally, when you pass the context object as an argument, you can provide a custom name for the context object inside the scope.
+
+<div class="sample" markdown="1" theme="idea">
+```kotlin
+fun writeToLog(str: String) {}
+
+fun main() {
+//sampleStart
+    "Hello".also { myString -> 
+        println("Also called on the string $myString")
+        writeToLog(myString)
+    }
+//sampleEnd
+}
+```
+</div>
+
+### Return value
+
+Scope functions return different values:
+* `apply` and `also` return the context object.
+* `let`, `run`, and `with` return the lambda result.
+
+These two options let you choose the proper function depending on what you do next in your code.
+
+Having the object itself as a return value is useful for chaining other operations on it.
```

Comment:
Not really. Chained operations can be simply put inside `apply`. Object as the return value is useful for returning the object from the function in which a scoped function has been used.

Reply:
>  Chained operations can be simply put inside apply.

This isn't idiomatic. If there's a chain of operations, both `apply` and `also` should be inside that chain, not outside.

Reply:
Added another example on using also when returning the context object from a function.